v-next/hardhat/src/internal/cli/main.ts

```diff
@@ -379,6 +381,11 @@ function parseDoubleDashArgs(
     const paramInfo = parametersMap.get(paramName);
 
     if (paramInfo === undefined) {
+      if (isGlobalParameter === true) {
```

Comment:
[Done in this commit](https://github.com/NomicFoundation/hardhat/pull/5373/commits/d299a671705ff843a5b1717e09b0760f1abde339)

Reply:
What if, instead of doing this, we enclose the call to `parseDoubleDashArguments` with a `try`/`catch` for global parameters and ignore that specific error? Something like:

```typescript
try {
  const parsedGlobalArgs = parseDoubleDashArgs(globalParamDefinitions, args);
  // ...
} catch (error) {
  if (!HardhatError.isHardhatError(error, HardhatError.ERRORS.ARGUMENTS.UNRECOGNIZED_NAMED_PARAM)) {
    throw error;
  }
}
```

Reply:
No, I don't think it's an elegant solution

Reply:
This is also more fragile, as a change in the error handling of a function would affect the product.